### DISCUSSION TITLE
Make bat Fetch lastest release from github repo when new version is released

### DISCUSSION THREAD

**Discussion by ZSHUser01** [tshakalekholoane/bat#17] [Ideas]
Hi! I wonder if could be possible to make bat binary update itself with the lastest release from the repo so users don't have to download the newer version and overwrite the older version with the latest version every time there's a new update for the app

Keep up the great work and thanks a lot for what you're doing

**Comment by tshakalekholoane** [comment]
Thank you. It is possible and certainly something that would improve the user experience. I just don't think that added convenience justifies the potential maintenance burden.

In any case, the distributions already have package managers that handle that sort of functionality. There's also the option of cloning and building the project locally which may or may not be as convenient as what you suggest depending on your level of familiarity with that.

**Reply by ZSHUser01** [reply]
Thank you for the exhaustive answer! I guess I'll just clone the repository and build the project locally as soon as a new version will be added in your repository as you suggested.

 Keep up the great work!

**Reply by tshakalekholoane** [reply]
Cheers.

In case you're unfamiliar or for anyone else wondering how to do that.

1. Clone the repository.

```sh
git clone https://github.com/leveson/bat.git
```

2. Whenever there are changes to the repository (you can subscribe to be notified of any changes using the watch button above), update your local repository by running the following command inside of the repository (which means you might need to run `cd bat/` first.

```sh
git pull
```

3. [Download Go](https://golang.org/dl/) and follow the [installation instructions](https://golang.org/doc/install) on the website. Alternatively, this could also be done through your package manager.

4. As outlined in the [README](https://github.com/leveson/bat/blob/master/README.md), run the following command to build the binary which will then appear your current directory.

```sh
go build ./cmd/bat
``` 

I am also considering making a GUI that might make supporting automatic updates easy so if that's something you might be interested in, have a look at this [discussion](https://github.com/leveson/bat/discussions/18).
